File: problems/866_prime_palindrome/solution.py
class Solution:
    def primePalindrome(self, n: int) -> int:
        def is_prime(x: int) -> bool:
            if x < 2:
                return False
            if x < 4:
                return True
            if x % 2 == 0 or x % 3 == 0:
                return False
            i = 5
            while i * i <= x:
                if x % i == 0 or x % (i + 2) == 0:
                    return False
                i += 6
            return True
        if 8 <= n <= 11:
            return 11
        for length in range(1, 6):
            for root in range(10**(length - 1), 10**length):
                s = str(root)
                pal = int(s + s[-2::-1])
                if pal >= n and is_prime(pal):
                    return pal
        return -1

File: problems/866_prime_palindrome/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    int primePalindrome(int n) {
        if (n >= 8 && n <= 11) return 11;
        for (int len = 1; len <= 5; len++) {
            for (int root = pow(10, len - 1); root < pow(10, len); root++) {
                string s = to_string(root);
                string pal = s + string(s.rbegin() + 1, s.rend());
                int num = stoi(pal);
                if (num >= n && isPrime(num)) return num;
            }
        }
        return -1;
    }
    bool isPrime(int x) {
        if (x < 2) return false;
        if (x < 4) return true;
        if (x % 2 == 0 || x % 3 == 0) return false;
        for (int i = 5; i * i <= x; i += 6)
            if (x % i == 0 || x % (i + 2) == 0) return false;
        return true;
    }
};
